src/ngraph/op/reverse.cpp

```diff
@@ -69,3 +70,116 @@ void op::Reverse::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVect
+                              static_cast<size_t>(rev_axes_rank),
+                              ").");
+
+        if (m_mode == "mask")
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                static_cast<size_t>(rev_axes_rank) == 1,
+                "The reversed_axes input rank needs to be equal to 1 in 'mask' mode.");
+
+            const auto rev_axes_mask_elems_count = static_cast<size_t>(rev_axes_shape[0]);
+            NODE_VALIDATION_CHECK(this,
+                                  rev_axes_mask_elems_count == static_cast<size_t>(input_rank),
+                                  "The number of elements in the reversed_axes tensor (",
+                                  rev_axes_mask_elems_count,
+                                  ") must match the input data tensor rank (",
+                                  static_cast<size_t>(input_rank),
+                                  ") in 'mask' mode.");
+        }
+    }
+
+    if (input_rank.is_static())
+    {
+        const auto rank = static_cast<size_t>(input_rank);
+        const auto rev_axes_node = input_value(1).get_node_shared_ptr();
+
+        if (rev_axes_node->is_constant())
+        {
+            const auto rev_axes_constant = dynamic_pointer_cast<op::Constant>(rev_axes_node);
+
+            if (m_mode == "index")
+            {
+                const AxisSet rev_axes = rev_axes_constant->get_axis_set_val();
+
+                NODE_VALIDATION_CHECK(this,
+                                      rev_axes.size() < rank,
+                                      "Too many axes(",
+                                      rev_axes,
+                                      ") have been provided for given input shape(",
+                                      input_shape,
+                                      ").");
+
+                bool all_axes_in_range = all_of(rev_axes.begin(),
+                                                rev_axes.end(),
+                                                [&rank](const size_t axis) { return axis < rank; });
+
+                NODE_VALIDATION_CHECK(this,
+                                      all_axes_in_range,
+                                      "Some of the provided axes (",
+                                      rev_axes,
+                                      ") are out of bounds (input rank: ",
+                                      static_cast<size_t>(input_rank),
+                                      ").");
+            }
+        }
+    }
+
+    set_output_type(0, get_input_element_type(0), input_shape);
+}
+
+shared_ptr<Node> op::v1::Reverse::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<op::v1::Reverse>(new_args.at(0), new_args.at(1), m_mode);
+}
+
+void op::v1::Reverse::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    const auto delta = deltas.at(0);
+
+    const auto x = input(0).get_source_output();
```

Comment:
You can use `input_value()` function here
```suggestion
    const auto x = input_value(0);
```